### PR TITLE
fix: order history filter checkbox selection

### PR DESCRIPTION
**Pull request by sss-bct**

### CODE REVIEW

File: components/page-templates/OrderHistoryTemplate/OrderHistoryTemplate.tsx [hunk 1]
```diff
@@ -55,13 +58,17 @@ const OrderHistoryTemplate = (props: OrderHistoryProps) => {
     const order = items?.find((orderItem) => orderItem?.id === id) as Order
     setSelectedOrder(order)
   }
-  const handleFilterApply = (selectedFilters: string) => changeFilters(selectedFilters)
+  const handleFilterApply = (selectedFilters: string) => {
+    changeFilters(selectedFilters)
+    queryClient.removeQueries(ordersKeys.all)
```
Comment (kibo-kevinwatts) [inline components/page-templates/OrderHistoryTemplate/OrderHistoryTemplate.tsx:63]: is there another way we can clear these queries? Driven by the hook/param changes instead?
Having the component clear the query doesn't seem the best approach

Comment (sss-bct) [reply components/page-templates/OrderHistoryTemplate/OrderHistoryTemplate.tsx]: Yes, we can do using param change as well. Made the changes for the same.

File: components/product-listing/FacetItemList/FacetItemList.tsx [hunk 2]
```diff
@@ -25,15 +25,16 @@ const FacetItemList = (props: FacetItemListProps) => {
   } = props
   const [factItemList, setFactItemList] = useState<FacetValue[]>(itemList)
 
-  const handleFacetSelection = (selectedFacetItem: string, isApplied: boolean) => {
-    itemList.filter((facet) => facet.filterValue === selectedFacetItem)[0].isApplied = !isApplied
-    setFactItemList(itemList)
-    const selectedFacetItems = facetGetters.getSelectedFacetItems(factItemList)
+  const handleFacetSelection = async (selectedFacetItem: string, isApplied: boolean) => {
+    const newItemList = [...itemList]
+    newItemList.filter((facet) => facet.filterValue === selectedFacetItem)[0].isApplied = !isApplied
```
Comment (amolpawar-kibo) [inline components/product-listing/FacetItemList/FacetItemList.tsx:30]: could you please add optional check (.?)?

File: hooks/queries/useUserOrderQueries/useUserOrderQueries.ts [hunk 3]
```diff
@@ -31,21 +36,15 @@ const getOrders = async (variables: { filter: string; startIndex: number; pageSi
...
   const {
     data = {},
     isLoading,
     isSuccess,
-  } = useQuery(ordersKeys.all, () => getOrders(variables), {
-    enabled: !!(filters?.length || (orderNumber && billingEmail)),
-    retry: 3,
+    isFetching,
+  } = useQuery(ordersKeys.all, () => getOrders(param), {
+    enabled: param.isRefetching,
```
Comment (amolpawar-kibo) [inline hooks/queries/useUserOrderQueries/useUserOrderQueries.ts:45]: could you please add optional check (.?)?

**Pull request closed without merging**